pilot/pkg/networking/grpcgen/lds.go

```diff
@@ -185,6 +222,40 @@ func buildInboundFilterChain(nameSuffix string, tlsContext *tls.DownstreamTlsCon
 	return out
 }
 
+// buildRBAC builds the RBAC config expected by gRPC.
+//
+// See: xds/interal/httpfilter/rbac
+//
+//
+// TODO: gRPC also supports 'per route override' - not yet clear how to use it, Istio uses path expressions instead and we don't generate
+// vhosts or routes for the inbound listener.
+//
+// For gateways it would make a lot of sense to use this concept, same for moving path prefix at top level ( more scalable, easier for users)
+// This should probably be done for the v2 API.
+//
+//
+//
+// nolint: unparam
+func buildRBAC(node *model.Proxy, push *model.PushContext, suffix string, context *tls.DownstreamTlsContext, a rbacpb.RBAC_Action, policies []model.AuthorizationPolicy) *rbacpb.RBAC {
```

Comment:
Can we reuse the authz builder, in case there are any changes in the future there?

Reply:
I don't think so, like LDS the generated output is slightly different.

However in future I am trying to define a subset of RBAC that can be used for both Envoy and proxyless, and works as  K8S Gateway policy attachment at route level ( supported by Envoy, not by current Istio API). The ideal is for this new API to be shared across all proxies ( and hopefully adopted by other gateway vendors ), and to work better with gateways ( O(1) vs O(N), etc).